src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2672,14 +2672,16 @@ bool primitive_inst::is_valid_fusion() const {
         const auto& outer_dep = _deps[outer_dep_idx];
 
         const auto& outer_dep_pshape = outer_dep.first->_impl_params->get_output_layout().get_partial_shape();
+        size_t outer_dep_pshape_count = outer_dep_pshape.is_static() ? ov::shape_size(outer_dep_pshape.to_shape()) : 0;
```

Comment:
Just curious, Do we need to check whether is_static()?

Reply:
Yes, `to_shape()` is only for static shapes.